docs/blog/2019-07-17-how-component-shadowing-works/index.md

```diff
@@ -0,0 +1,434 @@
+---
+title: How Shadowing Works
+path: /how-shadowing-works
+author: Chris Biscardi and John Otander
+date: 2019-07-17
+tags:
+  - open source
+  - webpack
+  - gatsby
+  - themes
+---
+
+Shadowing is a powerful feature that allows theme users to override components,
+objects, and anything else in a theme's `src` directory.
+
+> **Note** This is a technical deep dive into how Shadowing works. If you'd
+> like to learn about what Shadowing is, see the [What is Component Shadowing?](/blog/2019-04-29-component-shadowing/)
+> blog post.
+
+Shadowing works by using a [webpack resolver plugin](https://webpack.js.org/api/resolvers/)
+that maps themes in a `gatsby-config.js` to possible shadowed files. This gets
+especially mind melty because themes can add parent themes to a configuration so we
+need to be able to walk the composition of themes to determine the "last shadow"
+since the last shadowed theme file wins in the algorithm.
+
+## Theme Composition
+
+It's important to begin discussing how the composition of themes works. An
+end user of a theme can configure any number of themes. Each of these
+themes are considered sibling themes. Here is a `gatsby-config.js` that
+configures two sibling themes:
+
+```js:title=gatsby-config.js
+module.exports = {
+  _experimentalThemes: [
+    "gatsby-theme-tomato-blog",
+    "gatsby-theme-tomato-portfolio",
+  ],
+}
+```
+
+Both of the themes above (blog and portfolio) can install and configure
+any other theme so we end up with a tree of themes which we call a theme
+composition.
+
+The theme composition itself has a few properties:
+
+- the last theme wins
+- a theme that uses another theme is the child theme
+- a theme that is used by another theme is the parent theme
+- theme trees are flattened during resolution
+
+These characteristics are what we use in the component shadowing algorithm
+to decide which component to render. So, for example, if `gatsby-theme-tomato-blog`
+has `gastby-theme-parent` as a parent theme we'd result in the following themes
+array:
+
+```js
+const themesArray = [
+  "gatsby-theme-parent",
+  "gatsby-theme-tomato-blog",
+  "gatsby-theme-tomato-portfolio",
+]
+```
+
+This means that `gatsby-theme-tomato-portfolio` receives priority for
+component resolution.
```

Comment:
```suggestion
component resolution, because it is last in the array.
```

I think explicitly restating this helps drive home the first bullet point that "the last theme wins"